pkg/common/common.go

```diff
@@ -239,3 +241,56 @@ func GetImagePullPolicy(image string) (corev1.PullPolicy, error) {
 	}
 	return corev1.PullAlways, nil
 }
+
+// GenerateCliArgsFromConfigMap generates CLI arguments from a ConfigMap.
+//
+// This function takes a ConfigMap and a CLI subcommand(s), and returns a slice of strings representing
+// the CLI arguments from the subcommand(s) followed by the arguments from the ConfigMap.
+// The function processes each key-value pair in the ConfigMap as follows:
+//
+//  1. If the ConfigMaps' key starts with single '-' or double '--', it is left unchanged.
+//  2. If an argument key is passed without `-` or `--` prefix, then `--` is added as a key prefix.
+//  3. If the ConfigMap value is "true" or "false" (case-insensitive), it is converted to lowercase
+//     and used without single quotes surroundings (boolean value).
+//  4. The formatted key-value pair is added to the result that is alphabetically sorted.
+//
+// Args:
+//
+//		configMap: A pointer to a corev1.ConfigMap containing key-value pairs.
+//		cliSubCommand: The CLI subcommand(s) as a string, for example 'server'
+//	                or 'node-agent', 'server'
+//
+// Returns:
+//
+//	A slice of strings representing the CLI arguments.
+func GenerateCliArgsFromConfigMap(configMap *corev1.ConfigMap, cliSubCommand ...string) []string {
+
+	// cliSubCommand elements to be a separate items in args slice
+	args := make([]string, len(cliSubCommand))
+	copy(args, cliSubCommand)
+
+	var keyValueArgs []string
+	// Iterate through each key-value pair in the ConfigMap
+	for key, value := range configMap.Data {
+		// Ensure the key is prefixed by "--" if it doesn't start with "--" or "-"
+		if !strings.HasPrefix(key, "--") && !strings.HasPrefix(key, "-") {
```

Comment:
Checking for "--" is redundant here. If `!strings.HasPrefix(key, "-")` evaluates to `true`, then `!strings.HasPrefix(key, "--")` always will as well.

Reply:
Thanks, resolved.